spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java

```diff
@@ -192,6 +213,29 @@ public String getLog(String platformName, String taskId) {
 		return findTaskLauncher(platformName).getLog(taskId);
 	}
 
+	@Override
+	public void stopTaskExecution(Set<Long> ids) {
+		final Map<String, Object> auditData = new LinkedHashMap<>();
+		logger.info("Stopping {} task executions.", ids.size());
+		Set<TaskExecution> taskExecutions = getTaskExecutions(ids);
+		Set<Long> childTaskExecutionIds = this.dataflowTaskExecutionDao.findChildTaskExecutionIds(ids);
+		Set<TaskExecution> childTaskExecutions = getTaskExecutions(childTaskExecutionIds);
+		for (TaskExecution taskExecution : taskExecutions) {
```

Comment:
Should we check the current status of task execution before we try to cancel?

Reply:
Another good question.   I thought about this for a little bit.
While that would be useful, for a task that is currently running, but there are reported conditions where a pod is hung because of memory problems even though the task is complete.   In this case they want the pod deleted (which the stop for k8's effectively does).    So we would still allow for a stop request to be issued for those tasks that are marked `COMPLETE`.